### PR TITLE
Expose model flavors under `mlflow` so users don't have to import them

### PR DESCRIPTION
**Pull request by harupy**
Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

## What changes are proposed in this pull request?

Expose model flavors under `mlflow` so users don't have to import them.

Before:

```python
import mlflow

mlflow.sklearn.log_model(...)  # raises an error without `import mlflow.sklearn`
```

After:

```python
import mlflow

mlflow.sklearn.log_model(...)  # works
```

## How is this patch tested?

Unit test

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

All model flavors are exposed under `mllfow`.

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [x] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/server-infra`: MLflow server, JavaScript dev server
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Review by tomasatdatabricks — COMMENTED** [review]
Will this require users to install dependencies (e.g. tensorflow) for all flavors to import mlflow?

**Comment by harupy** [comment]
@tomasatdatabricks
No, they don't have to because all the import statements are hidden in functions.

**Comment by harupy** [comment]
Our CI doesn't have a check that verifies `mlflow` still can be imported without flavor dependencies.

**Comment by tomasatdatabricks** [comment]
Hmm, it looks like mlflow.tensorflow.py does import tensorflow and keras at the top of the module. So we probably need to fix that and add a test as you mentioned. Otherwise looks good!

**Comment by harupy** [comment]
> mlflow.tensorflow.py does import tensorflow and keras at the top of the module.

Which lines?

**Comment by tomasatdatabricks** [comment]
https://github.com/mlflow/mlflow/blob/master/mlflow/tensorflow.py#L25-L26
https://github.com/mlflow/mlflow/blob/master/mlflow/tensorflow.py#L29=
+ there are other suspicious imports such as  e.g. contextlib and gorilla?

**Comment by harupy** [comment]
> https://github.com/mlflow/mlflow/blob/master/mlflow/tensorflow.py#L25
https://github.com/mlflow/mlflow/blob/master/mlflow/tensorflow.py#L29

These two import statements has been moved into functions.

**Comment by harupy** [comment]
> there are other suspicious imports such as e.g. contextlib and gorilla

- contextlib should be fine because it's a standard library.
-

**Comment by tomasatdatabricks** [comment]
@harupy Sorry I missed the tf changes! Yeah in that case it sounds good as long as we have a test.

Thanks for doing this!

File: .github/workflows/master.yml [hunk 1]
```diff
@@ -183,6 +183,20 @@ jobs:
...
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: '3.6'
+      - name: Install mlflow
+        run: |
+          pip install -e .
+      - name: Verify mlflow can be imported without errors
+        run: |
+          python -c "import mlflow"
```
Comment (harupy) [inline .github/workflows/master.yml:198]: @tomasatdatabricks Added a new job to verify `mlflow` can be installed without errors.

**Review by tomasatdatabricks — APPROVED** [review]
lgtm!

File: .github/workflows/master.yml (hunk 1 above)
Comment (tomasatdatabricks) [reply .github/workflows/master.yml:198]: Thanks!

**Comment by harupy** [comment]
Hmm, the storage size for GitHub free account is only 500MB ([link](https://docs.github.com/en/github/setting-up-and-managing-billing-and-payments-on-github/about-billing-for-github-actions#about-billing-for-github-actions)). Our wheel size is about 15 MB so we could instantly reach this limit.

**Pull request merged**